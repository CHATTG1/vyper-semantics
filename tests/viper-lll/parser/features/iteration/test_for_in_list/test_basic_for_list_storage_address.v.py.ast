%pgm(,
  %svdecl(addresses, %listT(%address, 3), %private),
  %fdecl(%@public, set, %param(i, %num) %param(val, %address), %void,
    %assign(%subscript(%svar(addresses), %var(i)), %var(val)))
  %fdecl(%@public, ret, %param(i, %num), %address,
    %return(%subscript(%svar(addresses), %var(i))))
  %fdecl(%@public, iterate_return_second, , %address,
    %assign(%var(count), 0)
    %forlist(i, %svar(addresses),
      %augassign(+=, %var(count), 1)
      %if(%compareop(%eq, %var(count), 2),
        %return(%var(i)))))
)
