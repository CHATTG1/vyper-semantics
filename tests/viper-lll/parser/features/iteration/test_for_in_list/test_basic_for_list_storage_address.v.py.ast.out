$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 1400437809 ) , $%seq ( $calldatacopy ( 320 , 4 , 64 ) , $assert ( $iszero ( $callvalue ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $uclamplt ( $calldataload ( $add ( 4 , 32 ) ) , $mload ( 32 ) ) , $sstore ( $add ( $sha3_32 ( 0 ) , $uclamplt ( $mload ( 320 ) , 3 ) ) , $mload ( 352 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 1657485817 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $%seq ( $mstore ( 0 , $sload ( $add ( $sha3_32 ( 0 ) , $uclamplt ( $mload ( 320 ) , 3 ) ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 3878364963 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $mstore ( 320 , 0 ) , $%seq ( $repeat ( 384 , 0 , 3 , $%seq ( $mstore ( 352 , $sload ( $add ( $sha3_32 ( 0 ) , $mload ( 384 ) ) ) ) , $%seq ( $with ( $_mloc , 320 , $mstore ( $_mloc , $clamp ( $mload ( 96 ) , $add ( $mload ( $_mloc ) , 1 ) , $mload ( 64 ) ) ) ) , $if ( $eq ( $mload ( 320 ) , 2 ) , $%seq ( $%seq ( $mstore ( 0 , $mload ( 352 ) ) , $return ( 0 , 32 ) ) ) ) ) ) ) ) , $stop ) ) ) , 0 ) ) )
